File: Codeforces/158A - Next Round.cpp
// ==================================================
// Problem  :   158A - Next Round
// Run time :   0.060 sec.
// Language :   C++11
// ==================================================

#include <cstdio>
using namespace std;

int main()
{
    int n, k;

    while(~scanf("%d %d", &n, &k)) {
        int scores[n];

        for(int i = 0; i < n; ++i)
            scanf("%d", scores+i);

        int i = --k;

        if(scores[k]) {
            while(++i < n && scores[k] == scores[i]);
        }
        else {
            while(!scores[--i]);
            ++i;
        }

        printf("%d\n", i);
    }

    return 0;
}

File: Codeforces/158A - Next Round.py
# ==================================================
# Problem  :   158A - Next Round
# Run time :   0.154 sec.
# Language :   Python 3
# ==================================================

def main():
	from sys import stdin, stdout

	inp = [int(x) for x in stdin.read().split()]

	n = inp.pop(0)
	k = inp.pop(0)

	k_th = inp[k-1]
	cnt = 0

	for x in inp:
		if x and x >= k_th:
			cnt += 1;

	stdout.write(str(cnt) + '\n')

if __name__ == '__main__':
	main()
